include/buffer.h

```diff
@@ -53,53 +61,24 @@ class buffer {
 
 	~buffer() {}
 
-	template <cl::sycl::access::mode Mode, typename Functor>
-	accessor<DataT, Dims, Mode, cl::sycl::access::target::global_buffer> get_access(handler& cgh, Functor rmfn) const {
-		return get_access<Mode, cl::sycl::access::target::global_buffer>(cgh, rmfn);
+	template <cl::sycl::access_mode Mode, typename Functor>
+	auto get_access(handler& cgh, Functor rmfn) const {
```

Comment:
It would be better to have explicit return types here.

Reply:
done in new commit.